pyiceberg/table/snapshots.py

```diff
@@ -255,6 +255,14 @@ def manifests(self, io: FileIO) -> List[ManifestFile]:
         """Return the manifests for the given snapshot."""
         return list(_manifests(io, self.manifest_list))
 
+    def data_manifests(self, io: FileIO) -> List[ManifestFile]:
+        """Return the data manifests for the given snapshot."""
+        return [manifest for manifest in self.manifests(io) if manifest.content == ManifestContent.DATA]
+
+    def delete_manifests(self, io: FileIO) -> List[ManifestFile]:
+        """Return the delete manifests for the given snapshot."""
+        return [manifest for manifest in self.manifests(io) if manifest.content == ManifestContent.DELETES]
+
```

Comment:
Might be cleaner to just allow passing an optional `content_filter` to the `manifest` method instead of exposing two new methods

Reply:
yep, definitely cleaner